cleanSweStore.pl

```diff
@@ -0,0 +1,169 @@
+	   'man'=>\$man,
+	   'projects=s' => \$project, 
+	   'execute!' => \$execute,
+    	   'debug' => \$debug,
+      	    ) or pod2usage(-verbose => 0);
+
+pod2usage(-verbose => 1)  if ($help);
+pod2usage(-verbose => 2)  if ($man);
+unless (defined($project)) {
+	print "You must provide a list of projects to clean, format: runfoldername\tprojectid\n";
+	pod2usage(-verbose => 1);
+	exit;
+}
+
+#
+# Save all runfolder and the associated projects into a hash structure
+#
+# var => {
+#	runfolder_name1 => {
+#		projId1 => 1,
+#		projId2 => 1,
+#	}
+#	runfolder_name2 => {
+#		projId3 => 1,
+#	        projId4 => 1,
+#	}#
+# }
+#
+open PROJECTS, $project or die "Couldn't open project file: $project!";
+my $dataToClean;
+while(<PROJECTS>){
+	if(!/^#/) {
+		chomp;
+		my ($runfolder,$project) = split(/\t/, $_);
+		$project =~ s/\s+//;
+		$dataToClean->{$runfolder}->{$project} = 1;
+	}
+}
+close(PROJECTS);
+
+my $timestamp = time;
+
+open REMOVED, "> removedFromSweStore.$timestamp.log" or die "Couldn't open output file: removedFromSweStore.$timestamp.log!\n";
+open SAVED, "> leftOnSweStore.$timestamp.log" or die "Couldn't open output file: leftOnSweStore.$timestamp.log!\n";
+
+print "Cleaning swestore!\n";
+#
+# Remove the provided projects from SweStore
+#
+foreach my $runfolder (keys %{$dataToClean}) { # Process each runfolder 
+	my ($year,$month,$day) = ($runfolder =~ m/^(\d{2})(\d{2})(\d{2})_[A-Z0-9]+_[0-9]+_[A-Z0-9]/);
+	# Find each  stored project at SweStore, for the specified runfolder
+	my $projects = qx(ils $swestorePath/20$year-$month/$runfolder/Projects/);
+	#Result from ils
+	#
+	# ils /ssUppnexZone/proj/a2009002/2014-06/140605_D00118_0144_AC44G7ACXX/Projects/
+	# /ssUppnexZone/proj/a2009002/2014-06/140605_D00118_0144_AC44G7ACXX/Projects:
+	# C- /ssUppnexZone/proj/a2009002/2014-06/140605_D00118_0144_AC44G7ACXX/Projects/MK-0401
+	# C- /ssUppnexZone/proj/a2009002/2014-06/140605_D00118_0144_AC44G7ACXX/Projects/MK-0429
+	#
+	my @projectPath = split(/\n/,$projects);
+	my %foundProjects;
+	# Only extract information from lines containing "C-"
+	foreach (@projectPath) {
+		if(/^[ ]*C-/) {
+			my ($project) = ($_ =~ m/.*\/(.*)$/);
+			$project =~ s/\s+//;
+			$foundProjects{$project} = 1;
+		}
+	}
+	# Calculate number of projects found
+	my $numFoundProjects = (keys %foundProjects);
+	# Calculate number of projects that have been provided
+	my $numRemoveProjects = (keys %{$dataToClean->{$runfolder}});
+	#Make sure that the provided projects exist on SweStore, if not terminate the script.
+	foreach my $key (keys %{$dataToClean->{$runfolder}}) {
+		if(!exists($foundProjects{$key})) {
+			die "Couldn't find project $key on swestore: /ssUppnexZone/proj/a2009002/20" . $year . '-' . $month . '/' . $runfolder . "/Project\n";
+		}
+	}
+	
+	if($numFoundProjects < $numRemoveProjects) { #Cannot remove more projects than found on SweStore
+		die "You can't remove more projects than what exists, found $numFoundProjects, removing $numRemoveProjects!\n";
```

Comment:
if i'm not mistaken, this check is redundant to the check above and will actually never be true